### PR TITLE
Sign In and Sign Up with Github

### LINKED ISSUE #2: As a user, I want to be able to sign up/sign in to the system with my github account
Right after sign-up, a user should be asked to enter his name and surname, and group

### PR DESCRIPTION
**Pull request by CristianB13**

### CODE REVIEW

File: MyClassroom.Application/Services/IUserService.cs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+﻿using MyClassroom.Core.Models;
+
+namespace MyClassroom.Application.Services
+{
+    public interface IUserService
+    {
+        Task<User?> AddRole(int userId, int roleId);
+        Task<User?> AddRole(int userId, string roleName);
+        Task<User> AddUser(User user);
+        Task<User?> GetUserByIdAsync(int id);
```
Comment (bartrod) [inline MyClassroom.Application/Services/IUserService.cs:10]: Let's try to bo consistent. If you want to use Async postfix please use it in all the methods.

File: MyClassroom.Application/Services/UserService.cs [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+            return user;
+        }
+
+        public async Task<User?> AddRole(int userId, int roleId)
+        {
+            var userDb = await _context.Users.Include(u => u.Roles).SingleOrDefaultAsync(u => u.Id == userId);
+            var roleDb = await _context.Roles.SingleOrDefaultAsync(r => r.Id == roleId);
+            if (userDb is null || roleDb is null)
+            {
+                return null;
```
Comment (bartrod) [inline MyClassroom.Application/Services/UserService.cs:36]: What will happen if we will return null? Will it be reflected to the action on the application?

File: MyClassroom/Views/Shared/_Layout.cshtml [hunk 3]
```diff
@@ -25,6 +25,15 @@
                         <li class="nav-item">
                             <a class="nav-link text-dark" asp-area="" asp-controller="Home" asp-action="Privacy">Privacy</a>
                         </li>
+                        @if (!User.Identity?.IsAuthenticated ?? true)
```
Comment (bartrod) [inline MyClassroom/Views/Shared/_Layout.cshtml:28]: Are we going to support log-out operation?

**Review by bartrod — APPROVED** [review]
I left few questions

File: MyClassroom/Views/Shared/_Layout.cshtml (hunk 3 above)
Comment (CristianB13) [reply MyClassroom/Views/Shared/_Layout.cshtml:28]: Yes, I will have to implement logout as well

File: MyClassroom.Application/Services/IUserService.cs (hunk 1 above)
Comment (CristianB13) [reply MyClassroom.Application/Services/IUserService.cs:10]: Yes, I will change them

File: MyClassroom.Application/Services/UserService.cs (hunk 2 above)
Comment (CristianB13) [reply MyClassroom.Application/Services/UserService.cs:36]: Yes, I think I should throw an exception here instead of returning null

**Pull request merged**